src/include/attribute.h

```diff
@@ -189,7 +189,7 @@ typedef struct attribute attribute;
 struct attribute_def {
 	char	*at_name;
 	int	(*at_decode)(attribute *patr, char *name, char *rn, char *val);
-	int	(*at_encode)(attribute *pattr, pbs_list_head *phead,
+	int	(*at_encode)(attribute *pattr, pbs_list_node *phead,
```

Comment:
I went through the change and I do not like the renaming of pbs_list_head to pbs_list_node. In current name, i.e., pbs_list_head actually is more readable, and tells me that we are passing a pointer to the head of the linked list, and not some random node in the linked list. Of course, the variable itself is called phead (in most places) signifying that it is indeed the head of the linked list, but having the type itself point that out is actually more readable (ie, if you are looking at prototypes which do not include the parameter name).

In fact the current code defines two types to make things more readable…
typedef struct pbs_list_link {
struct pbs_list_link *ll_prior;
struct pbs_list_link *ll_next;
void	*ll_struct;
} pbs_list_link;
typedef pbs_list_link pbs_list_head;

Reply:
I agree that knowing the head node in a list is a good thing. However I believe using variable name to indicate that is a better option. Having two type names for the same structure can cause confusion to new developers.